### PR TITLE
xz: 5.4.6 -> 5.6.2

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

https://github.com/tukaani-project/xz/releases/tag/v5.6.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by trofi — APPROVED** [review]

File: pkgs/tools/compression/xz/default.nix [hunk 1]
```diff
@@ -11,14 +11,11 @@
...
-  version = "5.4.6"; # Beware of CVE-2024-3094 and related risks!!!
+  version = "5.6.2";
 
   src = fetchurl {
-    url = with finalAttrs;
-      # The original URL has been taken down.
-      # "https://github.com/tukaani-project/xz/releases/download/v${version}/xz-${version}.tar.bz2";
-      "mirror://sourceforge/lzmautils/xz-${version}.tar.bz2";
-    sha256 = "sha256-kThRsnTo4dMXgeyUnxwj6NvPDs9uc6JDbcIXad0+b0k=";
+    url = with finalAttrs; "https://github.com/tukaani-project/xz/releases/download/v${version}/xz-${version}.tar.xz";
```
Comment (consistency123) [inline pkgs/tools/compression/xz/default.nix:17]: Why is this .tar.xz while the [other pull request](https://github.com/NixOS/nixpkgs/pull/315842) is .tar.bz2?

**Review by consistency123 — CHANGES_REQUESTED** [review]
Why are we still not using [fetchFromGitHub](https://ryantm.github.io/nixpkgs/builders/fetchers/#fetchfromgithub)?

```
src = fetchFromGitHub {
  owner = "tukaani-project";
  repo = "xz";
  rev = "v${version}"; / rev = "3ec664d3f652133136587a51d4505b1abe1acdd7";
  hash = ""
};
```

**Comment by trofi** [comment]
> Why are we still not using [fetchFromGitHub](https://ryantm.github.io/nixpkgs/builders/fetchers/#fetchfromgithub)?

`xz` uses `autoconf` build system and is one of the earliest bootstrap packages. That would pull in `autoreconfHook` with it's dependencies like `perl` and might produce hard to resolvecircular dependencies. That also would increase bootstrap closure and slow down the initial `stdenv` build. Did you try to make it work?

**Comment by trofi** [comment]
Needs a minor conflict resolve.

**Review by JohnRTitor — APPROVED** [review]
Hopefully there is no backdoor now

**Pull request merged**